misc/futures-bounded/src/stream_set.rs

```diff
@@ -0,0 +1,60 @@
+use futures_util::stream::BoxStream;
+use futures_util::Stream;
+use std::task::{ready, Context, Poll};
+use std::time::Duration;
+
+use crate::{PushError, StreamMap, Timeout};
+
+/// Represents a list of [Stream]s.
```

Comment:
```suggestion
/// Represents a set of [Stream]s.
```

In case we stick with `StreamSet`.